.cloudbuild/ci/unit-tests-linux.yaml

```diff
@@ -5,7 +5,15 @@ steps:
     secretEnv:
       - TELEPORT_ENTERPRISE_LICENSE
     entrypoint: /bin/bash
-    args: ["-c", "make test"]
+    args:
+      - "-c"
+      - |
+        curl -fsSL https://apt.releases.hashicorp.com/gpg | apt-key add - &&
+        apt update && apt install -y software-properties-common &&
+        apt-add-repository "deb [arch=amd64] https://apt.releases.hashicorp.com $(lsb_release -cs) main" &&
+        apt update && apt install -y terraform &&
```

Comment:
I'd strongly recommend pinning the terraform version.  For apt:

```
apt install -y terraform=1.1.9
```
(or 1.2.2 for the most recent version -- I use an older one to show that it works)


Alternatively:
```
curl -o terraform.zip https://releases.hashicorp.com/terraform/${TF_VERS}/terraform_${TF_VERS}_linux_386.zip
unzip terraform.zip
mv terraform /usr/bin/terraform 
```

Without pinning we may see tests start to fail in this repo due to upstream changes. These will then block completely unrelated PRs and scatterblast the work of fixing current code to match the latest upstream.  With the version pinned, the necessary local changes can be added in the same PR that updates the pin.

Reply:
Thank you
You are totally right, it could break for unrelated changes
I will pin the latest version (1.2.2)

What do you think should be the makefile check for Terraform Version?
Should I check for the same minor we use in the cicd?

It used to check for `Terraform v1.1`, but I changed to only check for `Terraform v1.`
But now I wonder, if we should be more strict and match against `Terraform v1.2`

Reply:
> What do you think should be the makefile check for Terraform Version?
> Should I check for the same minor we use in the cicd?

Matching the minor version sounds reasonable.  Ideally we'd force the exact version, but I don't think we have strong enough automation around the toolchain to expect this for all developers.

> Thank you You are totally right, it could break for unrelated changes I will pin the latest version (1.2.2)
> 
> What do you think should be the makefile check for Terraform Version? Should I check for the same minor we use in the cicd?
> 
> It used to check for `Terraform v1.1`, but I changed to only check for `Terraform v1.` But now I wonder, if we should be more strict and match against `Terraform v1.2`

I think matching minor version is a good compromise for now.  Ideally we'd match the full version, but we don't have good toolchain management in place for this yet, and I don't want to scope creep this improvement to solve that more difficult problem.